### PR TITLE
Improved the tests as requested in #2

### PR DESCRIPTION
**Pull request by flyhard**
Went over the tests as requested in #2 : 
- fixed a bug in help.go that made the result unstable. The example_test.go would not pass in Docker but locally it would pass (I guess MAC vs linux did something)
- enforced running the tests in the Docker build
- improved build time of the Dockerfile by downloading the dependencies before adding the rest of the files
- added t.Run(...) to the table tests so the results are nicer displayed.
- added testcases for Hashes
- cleaned the go.mod file by running go mod tidy
- ran gofmt on the modified files

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 
 RUN mkdir -p /app
-COPY . /app
 WORKDIR /app
-RUN go get
+COPY go.mod /app/
+COPY go.sum /app/
+RUN go mod download
+
+COPY . /app
```
Comment (vertoforce) [inline Dockerfile:9]: Nice I like this, I think I'll do that from now on in my Dockerfiles :)

File: ioc/example_test.go [hunk 2]
```diff
@@ -3,13 +3,20 @@ package ioc
...
 	"fmt"
+	"sort"
 	"strings"
 )
 
 func ExampleGetIOCs() {
 	data := `this is a bad url http[://]google[.]com/path`
 
 	iocs := GetIOCs(data, false, true)
+	sort.SliceStable(iocs, func(i, j int) bool {
```
Comment (vertoforce) [inline ioc/example_test.go:14]: What exactly is this doing? Sorting the IOCs?

File: ioc/extra_test.go [hunk 3]
```diff
@@ -114,9 +119,16 @@ func BenchmarkGetIOCsFromHTML(b *testing.B) {
 		articles = append(articles, fileContents)
 		return nil
 	})
+	if err != nil {
```
Comment (vertoforce) [inline ioc/extra_test.go:122]: Good call, thank you :)

File: ioc/help.go [hunk 4]
```diff
@@ -10,20 +12,10 @@ func uniqueStringSlice(slice []string) []string {
...
-			// duplicate element, move on
-			continue
-		}
-		copy[i+1] = copy[j]
-		i++
-	}
+	// Duplicate and sort
+	strings := slice
+	sort.Strings(strings)
+	unique.Strings(&strings)
```
Comment (vertoforce) [inline ioc/help.go:19]: Clean! Love it.

File: ioc/ioc_test.go [hunk 5]
```diff
@@ -182,11 +188,15 @@ func TestGetIOCs(t *testing.T) {
 	}
 
 	for _, test := range testsStandardizedDefangs {
-		if iocs := GetIOCs(test.input, true, true); !reflect.DeepEqual(iocs, test.want) {
-			t.Errorf("[standardizedDefang=true] IOCType(%q), found %v =/= wanted %v", test.input, iocs, test.want)
-		}
-	}
+		t.Run(test.input, func(t *testing.T) {
```
Comment (vertoforce) [inline ioc/ioc_test.go:191]: So what's the advantage of doing this `t.Run` format instead of the way I did it?

Comment (vertoforce) [reply ioc/ioc_test.go:191]: Does it break out the tests more in the output or something?

**Review by vertoforce — APPROVED** [review]
Just a couple questions, but overall looks good :)

File: ioc/example_test.go (hunk 2 above)
Comment (flyhard) [reply ioc/example_test.go:14]: It makes the result deterministic. Without it the IOCs come out in different order dependent on the OS you run them on - and that breaks tests.

File: ioc/ioc_test.go (hunk 5 above)
Comment (flyhard) [reply ioc/ioc_test.go:191]: Yes, it does. When a test fails you see the name (here test.input) of the case that failed. Try breaking a test and see for yourself. Most IDEs support it as well (intelliJ in my case)

File: ioc/example_test.go (hunk 2 above)
Comment (vertoforce) [reply ioc/example_test.go:14]: Oh good to know, thanks!

**Pull request merged**